README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# liberia: Datasets for Use in Designing Surveys in Liberia <img src="man/figures/logo.svg" width="200" align="right" />

<!-- badges: start -->

[![Project Status: Active – The project has reached a stable, usable
state and is being actively
developed.](https://www.repostatus.org/badges/latest/active.svg)](https://www.repostatus.org/#active)
[![R-CMD-check](https://github.com/spatialworks/liberia/workflows/R-CMD-check/badge.svg)](https://github.com/spatialworks/liberia/actions)
[![CodeFactor](https://www.codefactor.io/repository/github/spatialworks/liberia/badge)](https://www.codefactor.io/repository/github/spatialworks/liberia)
<!-- badges: end -->

Designing surveys require relevant datasets to be used as basis for
sample size calculations, sampling design, survey planning/logistics and
survey implementation. These include datasets on population, lists of
sampling clusters, map datasets for spatial sampling, and previous
survey datasets that can be used for estimating indicator variance and
design effects. This package contains relevant datasets for use in
designing surveys in Liberia.

## Installation

Install `liberia` package from GitHub via \`remotes\`\`:

``` r
if (!require(remotes)) install.packages("remotes")
remotes::install_github("spatialworks/liberia")
```

## Usage

### Counties map

``` r
plot(sf::st_geometry(counties))
```

<img src="man/figures/README-liberia1-1.png" style="display: block; margin: auto;" />

### Districts map

``` r
plot(sf::st_geometry(districts))
```

<img src="man/figures/README-liberia2-1.png" style="display: block; margin: auto;" />

### Clans map

``` r
plot(sf::st_geometry(clans))
```

<img src="man/figures/README-liberia3-1.png" style="display: block; margin: auto;" />

### Locality population

``` r
locality
#> # A tibble: 3,412 × 9
#>      FID County      District  Clan    `Locality/ Town`  MALE FEMALE TOTAL    HH
#>    <dbl> <chr>       <chr>     <chr>   <chr>            <dbl>  <dbl> <dbl> <dbl>
#>  1  3555 Grand Bassa District… Kpogbl… Cotton Tree          7      4    11     4
#>  2  3556 Grand Bassa District… L A C … 10 Dollars Camp     35     42    77    15
#>  3  3557 Grand Bassa District… L A C … 15 Houses Camp      91     86   177    21
#>  4  3558 Grand Bassa District… L A C … 3 Old Camp          62     56   118    38
#>  5  3559 Grand Bassa District… L A C … 3.2 Office           4      2     6     1
#>  6  3560 Grand Bassa District… L A C … 5 House Camp        25     26    51    11
#>  7  3561 Grand Bassa District… L A C … 6 Houses Camp      162    135   297    57
#>  8  3562 Grand Bassa District… Faitro  Adolphus           118    114   232    46
#>  9  3563 Grand Bassa District… Gaye P… Alex Johnson        27     28    55     8
#> 10  3564 Grand Bassa District… Zuzohn  Alex Village         5      2     7     3
#> # … with 3,402 more rows
```

### Enumeration areas population

``` r
enumerationArea
#> # A tibble: 751 × 8
#>    County      District     Clan                 EACODE  MALE FEMALE    HH TOTAL
#>    <chr>       <chr>        <chr>                 <dbl> <dbl>  <dbl> <dbl> <dbl>
#>  1 Grand Bassa Commonwealth Upper Buchanan Co…   9.12e8   119    123    44   242
#>  2 Grand Bassa Commonwealth Gorzohn Community    9.12e8   160    174    70   334
#>  3 Grand Bassa Commonwealth Central Buchanan     9.12e8   147    158    53   305
#>  4 Grand Bassa Commonwealth Korkorwein/Old Ba…   9.12e8   196    189    56   385
#>  5 Grand Bassa Commonwealth Central Buchanan     9.12e8   209    210    73   419
#>  6 Grand Bassa Commonwealth Central Buchanan     9.12e8   287    288   104   575
#>  7 Grand Bassa Commonwealth Gorzohn Community    9.12e8   265    301    98   566
#>  8 Grand Bassa Commonwealth Gorzohn Community    9.12e8   258    324   114   582
#>  9 Grand Bassa Commonwealth Gorzohn Community    9.12e8    86     99    31   185
#> 10 Grand Bassa Commonwealth Central Buchanan     9.12e8   205    169    66   374
#> # … with 741 more rows
```

### Monrovia population

``` r
monrovia
#> # A tibble: 166 × 6
#>    County      District         `Town/ Locality`     Male Female Total
#>    <chr>       <chr>            <chr>               <dbl>  <dbl> <dbl>
#>  1 Montserrado Greater Monrovia Bernard Quarters     1822   1739  3561
#>  2 Montserrado Greater Monrovia Grandcess Yard       3070   3470  6540
#>  3 Montserrado Greater Monrovia Johnsonville Road A  2677   3047  5724
#>  4 Montserrado Greater Monrovia Kaba Town            2939   3066  6005
#>  5 Montserrado Greater Monrovia Rock Spring Valley   1956   1850  3806
#>  6 Montserrado Greater Monrovia Soniwein             3293   3103  6396
#>  7 Montserrado Greater Monrovia Warwein               773    726  1499
#>  8 Montserrado Greater Monrovia 72nd Community       6883   7516 14399
#>  9 Montserrado Greater Monrovia A.B Tolbert Road     3195   3522  6717
#> 10 Montserrado Greater Monrovia Barchue Town         1561   1606  3167
#> # … with 156 more rows
```


NEWS.md
# liberia 0.0.0.9000

* Added a `NEWS.md` file to track changes to the package.





data-raw/mapdata.R
################################################################################
#
# Libraries
#
################################################################################
library(sf)
library(readxl)
library(stringr)


################################################################################
#
# Read maps
#
################################################################################

## Counties
#counties <- readOGR(dsn = "data-raw/maps/lbr_admbnda_adm1_ocha",
#                    layer = "lbr_admbnda_adm1_ocha",
#                    verbose = FALSE)

counties <- sf::st_read(
  dsn = "data-raw/maps/lbr_admbnda_adm1_ocha",
  layer = "lbr_admbnda_adm1_ocha"
)
usethis::use_data(counties, overwrite = TRUE)


## Districts
# districts <- readOGR(dsn = "data-raw/maps/lbr_admbnda_adm2_ocha",
#                      layer = "lbr_admbnda_adm2_ocha",
#                      verbose = FALSE)

districts <- sf::st_read(
  dsn = "data-raw/maps/lbr_admbnda_adm2_ocha",
  layer = "lbr_admbnda_adm2_ocha"
)
usethis::use_data(districts, overwrite = TRUE)

## Clans
#clans <- readOGR(dsn = "data-raw/maps/clans",
#                    layer = "LIB",
#                    verbose = FALSE)
#clans <- clans[ , 1:10]

clans <- sf::st_read(dsn = "data-raw/maps/clans", layer = "LIB")
usethis::use_data(clans, overwrite = TRUE)


## Settlements
#settlements <- readOGR(dsn = "data-raw/maps/lbr_plp_ocha",
#                       layer = "lbr_plp_ocha",
#                       verbose = FALSE)

settlements <- sf::st_read(
  dsn = "data-raw/maps/lbr_plp_ocha", layer = "lbr_plp_ocha"
)
usethis::use_data(settlements, overwrite = TRUE)

## Enumeration areas
# grandBassaEA <- readOGR(dsn = "data-raw/maps/Bassa_EA",
#                         layer = "Bassa_EA",
#                         verbose = FALSE)
#
# grandBassaEA <- spTransform(grandBassaEA,
#                             CRSobj = CRS(proj4string(counties)))
#
# devtools::use_data(grandBassaEA, overwrite = TRUE)

grandBassaEA <- sf::st_read(
  dsn = "data-raw/maps/Bassa_EA", layer = "Bassa_EA"
)
grandBassaEA <- sf::st_transform(grandBassaEA, crs = sf::st_crs(counties))
#sf::st_crs(grandBassaEA) <- 4326
usethis::use_data(grandBassaEA, overwrite = TRUE)

# greaterMonroviaEA <- readOGR(dsn = "data-raw/maps/Greater_Monrovia_EA",
#                              layer = "Greater_Monrovia_EA",
#                              verbose = FALSE)
#
# greaterMonroviaEA <- spTransform(greaterMonroviaEA,
#                                  CRSobj = CRS(proj4string(counties)))
#
# devtools::use_data(greaterMonroviaEA, overwrite = TRUE)

greaterMonroviaEA <- sf::st_read(
  dsn = "data-raw/maps/Greater_Monrovia_EA", layer = "Greater_Monrovia_EA"
)
greaterMonroviaEA <- sf::st_transform(
  greaterMonroviaEA, crs = sf::st_crs(counties)
)
#sf::st_crs(greaterMonroviaEA) <- 4326
usethis::use_data(greaterMonroviaEA,  overwrite = TRUE)

health_facilities <- read_excel(
  path = "data-raw/maps/141121-lr-health-care-facilities.xlsx",
  sheet = 2, skip = 1
)
names(health_facilities) <- str_remove(
  string = names(health_facilities), pattern = "#"
)





R/data.R
################################################################################
#
#' counties
#'
#' Administrative Level 1 (counties)  map of Liberia
#'
#' @format A SpatialPolygonsDataFrame with 15 features:
#' \describe{
#' \item{\code{OBJECTID}}{Object identifier}
#' \item{\code{CNTRY_CODE}}{Three-character ISO country code. For Liberia it is
#'     LBR.}
#' \item{\code{WHO_CODE}}{Three-character WHO country code. For Liberia it is
#'     LIB.}
#' \item{\code{Hrparent}}{HR Parent}
#' \item{\code{Shape_Leng}}{Length of shape}
#' \item{\code{Shape_Area}}{Area of shape}
#' \item{\code{admin1name}}{Name of county}
#' \item{\code{admin1Pcod}}{County postal code}
#' \item{\code{admin0name}}{Name of country}
#' \item{\code{admin0Pcod}}{Country postal code. LR for Liberia.}
#' \item{\code{Pcod1_iso3}}{County postal code using ISO3 code}
#' }
#'
#' @source \url{https://data.humdata.org/dataset/liberia-admin-level-2-boundaries}
#'
#
################################################################################
"counties"


################################################################################
#
#' districts
#'
#' Administrative Level 2 (districts)  map of Liberia
#'
#' @format A SpatialPolygonsDataFrame with 136 features:
#' \describe{
#' \item{\code{OBJECTID}}{Object identifier}
#' \item{\code{CNTRY_NAME}}{Country name}
#' \item{\code{CNTRY_CODE}}{Three-character ISO country code. For Liberia it is
#'     LBR.}
#' \item{\code{WHO_CODE}}{Three-character WHO country code. For Liberia it is
#'     LIB.}
#' \item{\code{Shape_Leng}}{Length of shape}
#' \item{\code{Shape_Area}}{Area of shape}
#' \item{\code{admin1name}}{Name of county}
#' \item{\code{admin1Pcod}}{County postal code}
#' \item{\code{admin2name}}{Name of district}
#' \item{\code{admin2Pcod}}{District postal code}
#' \item{\code{Pcod1_iso3}}{County postal code using ISO3 code}
#' \item{\code{Pcod2_iso3}}{District postal code using ISO3 code}
#' }
#'
#' @source \url{https://data.humdata.org/dataset/liberia-admin-level-2-boundaries}
#'
#
################################################################################
"districts"


################################################################################
#
#' clans
#'
#' Administrative Level 2 (clans) map of Liberia
#'
#' @format A SpatialPolygonsDataFrame with 305 features:
#' \describe{
#' \item{\code{ID}}{Object identifier}
#' \item{\code{LBL}}{Clan identifier}
#' \item{\code{FIP}}{Two character country code for Liberia}
#' \item{\code{MMT_ID}}{Three character country code for Liberia}
#' \item{\code{SHORT__FRM}}{Short form name for Liberia}
#' \item{\code{LONG_FRM}}{Long form name for Liberia}
#' \item{\code{ADM0}}{Administrative level 0 of Liberia (Country name)}
#' \item{\code{ADM1}}{Administrative level 1 of Liberia (County name)}
#' \item{\code{ADM2}}{Administrative level 2 of Liberia (District name)}
#' \item{\code{ADM3}}{Administrative level 3 of Liberia (Clan name)}
#' }
#'
#' @source Map Maker Ltd. (2007). Clans, Liberia, 2007 Map Maker, Ltd. Available
#' at: \url{http://purl.stanford.edu/vm535pw6013}.
#'
#
################################################################################
"clans"


################################################################################
#
#' grandBassaEA
#'
#' Enumeration area map of Grand Bassa county
#'
#' @format A SpatialPolygonsDataFrame with 468 features:
#' \describe{
#' \item{\code{CCNAME}}{County name}
#' \item{\code{DNAME}}{District name}
#' \item{\code{CLNAME}}{Clan name}
#' \item{\code{FEACODE}}{Full EA code}
#' \item{\code{SAREA}}{Area map label}
#' \item{\code{EFEACODE}}{Extended full EA code}
#' \item{\code{CCODE}}{County code}
#' \item{\code{DCODE}}{District code}
#' \item{\code{CLCODE}}{Clan code}
#' \item{\code{EACODE}}{Enumeration area code}
#' \item{\code{TOTAL}}{Total population}
#' \item{\code{MALE}}{Male population}
#' \item{\code{FEMALE}}{Female population}
#' \item{\code{HH}}{Total households}
#' \item{\code{CCODE1}}{Extended county code}
#' \item{\code{DCODE1}}{Extended district code}
#' \item{\code{CLCODE1}}{Extended clan code}
#' \item{\code{EACODE1}}{Extended enumeration area code}
#' }
#'
#' @source Liberia Institute of Statistics and Geo-Information Services
#'
#
################################################################################
"grandBassaEA"


################################################################################
#
#' greaterMonroviaEA
#'
#' Enumeration area map of Greater Monrovia District
#'
#' @format A SpatialPolygonsDataFrame with 1967 features:
#' \describe{
#' \item{\code{CCNAME}}{County name}
#' \item{\code{DNAME}}{District name}
#' \item{\code{CLNAME}}{Clan name}
#' \item{\code{FEACODE}}{Full EA code}
#' \item{\code{SAREA}}{Area map label}
#' \item{\code{EFEACODE}}{Extended full EA code}
#' \item{\code{CCODE}}{County code}
#' \item{\code{DCODE}}{District code}
#' \item{\code{CLCODE}}{Clan code}
#' \item{\code{EACODE}}{Enumeration area code}
#' \item{\code{TOTAL}}{Total population}
#' \item{\code{MALE}}{Male population}
#' \item{\code{FEMALE}}{Female population}
#' \item{\code{HH}}{Total households}
#' \item{\code{CCODE1}}{Extended county code}
#' \item{\code{DCODE1}}{Extended district code}
#' \item{\code{CLCODE1}}{Extended clan code}
#' \item{\code{EACODE1}}{Extended enumeration area code}
#' }
#'
#' @source Liberia Institute of Statistics and Geo-Information Services
#'
#
################################################################################
"greaterMonroviaEA"


################################################################################
#
#' settlements
#'
#' Map of settlement locations in Liberia
#'
#' @format A SpatialPointsDataFrame with 14013 features:
#' \describe{
#' \item{\code{OBJECTID}}{Object identifier}
#' \item{\code{FNAME}}{Settlement name}
#' \item{\code{CLASS}}{A numeric value specifying the settlement class. 1 = Capital
#'     of country; 2 = Capital of county; 3 = Capital of district; 0 = village}
#' \item{\code{DESCRIPTIO}}{A string value describing the settlement class based
#'     on \code{CLASS}}
#' \item{\code{COORD_X}}{Longitude coordinate of settlement}
#' \item{\code{COORD_Y}}{Latitude coordiante of settlement}
#' \item{\code{admin1name}}{Name of county}
#' \item{\code{admin1Pcod}}{County postal code}
#' \item{\code{admin2name}}{Name of district}
#' \item{\code{admin2Pcod}}{District postal code}
#' \item{\code{admin0name}}{Name of country}
#' \item{\code{admin0Pcod}}{Country postal code. LR for Liberia.}
#' \item{\code{rowcacode0}}{Three-character ISO country code}
#' \item{\code{Pcod1_iso3}}{County postal code using ISO3 code}
#' \item{\code{Pcod2_iso3}}{District postal code using ISO3 code}
#' }
#'
#' @source \url{https://data.humdata.org/dataset/liberia-settlements-0}
#'
#
################################################################################
"settlements"


################################################################################
#
#' locality
#'
#' Dataset of localities with population size for Grand Bassa and Rural
#' Montserrado
#'
#' @format A data.frame with 9 columns and 3412 rows
#' \describe{
#' \item{\code{FID}}{Locality ID}
#' \item{\code{County}}{Name of county}
#' \item{\code{District}}{Name of district}
#' \item{\code{Clan}}{Name of clan}
#' \item{\code{`Locality/ Town`}}{Name of locality/town}
#' \item{\code{MALE}}{Number of male population}
#' \item{\code{FEMALE}}{Number of female population}
#' \item{\code{TOTAL}}{Total population}
#' \item{\code{HH}}{Number of households}
#' }
#'
#' @source Liberia Institute of Statistics and Geo-Information Services
#'
#
################################################################################
"locality"


################################################################################
#
#' enumerationArea
#'
#' Dataset of enumeration areas with population size for Grand Bassa and Rural
#' Montserrado
#'
#' @format A data.frame with 8 columns and 751 rows
#' \describe{
#' \item{\code{County}}{Name of county}
#' \item{\code{District}}{Name of district}
#' \item{\code{Clan}}{Name of clan}
#' \item{\code{EACODE}}{Enumeration area code}
#' \item{\code{MALE}}{Number of male population}
#' \item{\code{FEMALE}}{Number of female population}
#' \item{\code{HH}}{Number of households}
#' \item{\code{TOTAL}}{Total population}
#' }
#'
#' @source Liberia Institute of Statistics and Geo-Information Services
#'
#
################################################################################
"enumerationArea"


################################################################################
#
#' monrovia
#'
#' Dataset of enumeration areas with population size for Greater Monrovia
#' District
#'
#' @format A data.frame with 8 columns and 751 rows
#' \describe{
#' \item{\code{County}}{Name of county}
#' \item{\code{District}}{Name of district}
#' \item{\code{Town/Locality}}{Town or locality name}
#' \item{\code{MALE}}{Number of male population}
#' \item{\code{FEMALE}}{Number of female population}
#' \item{\code{TOTAL}}{Total population}
#' }
#'
#' @source Liberia Institute of Statistics and Geo-Information Services
#'
#
################################################################################
"monrovia"

